### ISSUE TITLE
Glitched infinite bonsai pot, then kicked from server.

### ISSUE THREAD

**Issue by CyaniideDrive** [Tschipp/CarryOn#369]
### Expected Behavior
Able to place one bonsai pot back down after picking up.
When picking up items after dying to try and fix the issue, not to get kicked from my server.

### Actual Behavior
Can place infinite bonsai pots.
After dying and picking back up my inventory I get booted from the server instantly and whenever I try to join it.

### Steps to Reproduce
Pick up Bonsai Pot.
Place it many times.
Die.
Pick up inventory.

### Version of Minecraft, Carry On, Forge
Minecraft version 1.18.1
Carry On version 1.17.0.7
Since it may help as well Bonsai Trees version 3.0.4
Forge version 39.0.79

### Screenshots encouraged
Didn't think to take screenshots before getting kicked.

**Comment by CyaniideDrive** [comment]
The Server spit out this pile of errors when it happened.

net.minecraft.ReportedException: Ticking playerat net.minecraft.server.level.ServerPlayer.m_9240_([ServerPlayer.java:502](http://serverplayer.java:502/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.server.network.ServerGamePacketListenerImpl.m_9933_([ServerGamePacketListenerImpl.java:206](http://servergamepacketlistenerimpl.java:206/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.network.Connection.m_129483_([Connection.java:233](http://connection.java:233/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.server.network.ServerConnectionListener.m_9721_([ServerConnectionListener.java:142](http://serverconnectionlistener.java:142/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.server.MinecraftServer.m_5703_([MinecraftServer.java:882](http://minecraftserver.java:882/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.server.dedicated.DedicatedServer.m_5703_([DedicatedServer.java:328](http://dedicatedserver.java:328/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.server.MinecraftServer.m_5705_([MinecraftServer.java:808](http://minecraftserver.java:808/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.server.MinecraftServer.m_130011_([MinecraftServer.java:668](http://minecraftserver.java:668/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.server.MinecraftServer.m_177918_([MinecraftServer.java:258](http://minecraftserver.java:258/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]
at [java.lang.Thread.run](http://java.lang.thread.run/)([Thread.java:833](http://thread.java:833/)) [?:?]Caused by: java.lang.RuntimeException: Attempted to load class net/minecraft/client/multiplayer/ClientLevel for invalid dist DEDICATED_SERVERat net.minecraftforge.fml.loading.RuntimeDistCleaner.processClassWithFlags([RuntimeDistCleaner.java:71](http://runtimedistcleaner.java:71/)) ~[fmlloader-1.18.1-39.0.79.jar%2323!/:1.0]at cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins([LaunchPluginHandler.java:88](http://launchpluginhandler.java:88/)) ~[modlauncher-9.1.0.jar%235!/:?]at cpw.mods.modlauncher.ClassTransformer.transform([ClassTransformer.java:120](http://classtransformer.java:120/)) ~[modlauncher-9.1.0.jar%235!/:?]at cpw.mods.modlauncher.TransformingClassLoader.maybeTransformClassBytes([TransformingClassLoader.java:50](http://transformingclassloader.java:50/)) ~[modlauncher-9.1.0.jar%235!/:?]at cpw.mods.cl.ModuleClassLoader.readerToClass([ModuleClassLoader.java:110](http://moduleclassloader.java:110/)) ~[securejarhandler-1.0.1.jar:?]at cpw.mods.cl.ModuleClassLoader.lambda$findClass$16([ModuleClassLoader.java:213](http://moduleclassloader.java:213/)) ~[securejarhandler-1.0.1.jar:?]at cpw.mods.cl.ModuleClassLoader.loadFromModule([ModuleClassLoader.java:223](http://moduleclassloader.java:223/)) ~[securejarhandler-1.0.1.jar:?]at cpw.mods.cl.ModuleClassLoader.findClass([ModuleClassLoader.java:213](http://moduleclassloader.java:213/)) ~[securejarhandler-1.0.1.jar:?]at cpw.mods.cl.ModuleClassLoader.loadClass([ModuleClassLoader.java:130](http://moduleclassloader.java:130/)) ~[securejarhandler-1.0.1.jar:?]at java.lang.ClassLoader.loadClass([ClassLoader.java:520](http://classloader.java:520/)) ~[?:?]at com.davenonymous.bonsaitrees3.blocks.BonsaiPotBlockEntity.onDataLoaded([BonsaiPotBlockEntity.java:472](http://bonsaipotblockentity.java:472/)) ~[bonsaitrees3-3.0.4.jar%2357!/:3.0.4]at com.davenonymous.libnonymous.base.BaseBlockEntity.m_142466_([BaseBlockEntity.java:107](http://baseblockentity.java:107/)) ~[libnonymous-2.0.3.jar%23160!/:2.0.3]
at tschipp.carryon.common.event.ItemEvents.onItemDropped([ItemEvents.java:130](http://itemevents.java:130/)) ~[carryon-1.18.1-1.17.0.7.jar%2362!/:1.17.0.7]at net.minecraftforge.eventbus.ASMEventHandler_1593_ItemEvents_onItemDropped_EntityJoinWorldEvent.invoke(.dynamic) ~[?:?]at net.minecraftforge.eventbus.ASMEventHandler.invoke([ASMEventHandler.java:85](http://asmeventhandler.java:85/)) ~[eventbus-5.0.3.jar%232!/:?]at [net.minecraftforge.eventbus.EventBus.post](http://net.minecraftforge.eventbus.eventbus.post/)([EventBus.java:302](http://eventbus.java:302/)) ~[eventbus-5.0.3.jar%232!/:?]at [net.minecraftforge.eventbus.EventBus.post](http://net.minecraftforge.eventbus.eventbus.post/)([EventBus.java:283](http://eventbus.java:283/)) ~[eventbus-5.0.3.jar%232!/:?]at net.minecraft.world.level.entity.PersistentEntitySectionManager.m_157538_([PersistentEntitySectionManager.java:80](http://persistententitysectionmanager.java/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.world.level.entity.PersistentEntitySectionManager.m_157533_([PersistentEntitySectionManager.java:72](http://persistententitysectionmanager.java:72/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.server.level.ServerLevel.m_8872_([ServerLevel.java:836](http://serverlevel.java:836/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.server.level.ServerLevel.m_7967_([ServerLevel.java:751](http://serverlevel.java:751/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at com.integral.enigmaticlegacy.items.StorageCrystal.retrieveDropsFromCrystal([StorageCrystal.java:76](http://storagecrystal.java:76/)) ~[EnigmaticLegacy-2.16.1.jar%23118!/:2.16.1]at com.integral.enigmaticlegacy.entities.PermanentItemEntity.m_6123_([PermanentItemEntity.java:261](http://permanentitementity.java:261/)) ~[EnigmaticLegacy-2.16.1.jar%23118!/:2.16.1]at net.minecraft.world.entity.player.Player.m_36277_([Player.java:555](http://player.java:555/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.world.entity.player.Player.m_8107_([Player.java:522](http://player.java:522/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.world.entity.LivingEntity.m_8119_([LivingEntity.java:2249](http://livingentity.java:2249/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]
at net.minecraft.world.entity.player.Player.m_8119_([Player.java:234](http://player.java:234/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]at net.minecraft.server.level.ServerPlayer.m_9240_([ServerPlayer.java:439](http://serverplayer.java:439/)) ~[server-1.18.1-20211210.034407-srg.jar%23262!/:?]

**Comment by Porcy** [comment]
Same problem here. Was carrying a bonsai pot and able to place it over and over again. That makes ite very hard to empty your hands :/ I was able to right click on a drawer from functionnal storage and put the bonsai pot in it.

**Comment by BonaireDreams** [comment]
Just experience this same problem. Picked up a bonsai pot could place down multiple bonsai's and was unable to ever empty my hands. Had to drown myself to get rid of the pot I was carrying.

**Comment by Arovei** [comment]
Similar issue, I can log back in though. I killed myself to fix it, but i had to put something else in the "empty hand" slot on my hot bar because when i scrolled over it, it put the bonsai back in my hand and I had to kill myself again. Working on some other stuff maybe 20 minutes later and with a different hotbar slot the bonsai ended up back in my hand again... not sure how to fix this unless im now permanently bugged in this world.

MC - v1.18.2
Carryon - v1.17.0.8
Bonsai Trees 3 - v3.0.1
Forge - v40.1.51
